ros2bag/ros2bag/verb/record.py

```diff
@@ -52,14 +58,14 @@ def main(self, *, args):  # noqa: D102
         if os.path.isdir(uri):
             return "Error: Output folder '{}' already exists.".format(uri)
 
-        try:
-            os.makedirs(uri)
-        except:
-            return "Error: Could not create bag folder '{}'.".format(uri)
+        self.create_bag_directory(uri)
 
         if args.all:
             rosbag2_transport_py.record(uri=uri, storage_id=args.storage, all=True)
         elif args.topics and len(args.topics) > 0:
             rosbag2_transport_py.record(uri=uri, storage_id=args.storage, topics=args.topics)
         else:
             self._subparser.print_help()
+
+        if os.path.isdir(uri) and not os.listdir(uri):
```

Comment:
It seems to be there is a design flaw here. I believe there should be a function (exposed) which can check whether the specified plugin is correct. That is, before calling straight up `record`, I think we should be able to initialize and setup the recorder at first and then call record. Having the recorder setup first allows for error checking.

Then second, the code for removing the folder only works right now because record has to be cancelled by `ctrl-c` (because of the spin inside). I believe this should be handled by a signint handler, to cancel the record deliberately.

Reply:
IMHO the "design flaw" is that the directory creation happens already in the python layer before instead of having the storage taking care of that. The reason for this is that cross platform file handling is hard to do.
Creating a separate `initialize` method would only solidify this workaround and would still leave some scenarios uncovered that cannot be detected beforehand.

Reply:
Regarding the second point I agree that we will need to do our own signal handling for that and then we also need to change this "failure detection". I would prefer to do this with another pull request.